lib/custom_set_cases.rb

```diff
@@ -0,0 +1,85 @@
+class CustomSetCase < OpenStruct
+  def name
+    'test_%s' % description.gsub(/ |-/, '_')
+  end
+
+  def test_body
+    instance_eval section
+  end
+
+  def union
+    "set1 = CustomSet.new #{set1}
+    set2 = CustomSet.new #{set2}
+    expected = CustomSet.new(#{expected})
+    #{assert_or_refute}_equal set1.union(set2), expected"
```

Comment:
I like how you're assigning expected to a variable, but it appears that you have expected and actual inverted which will cause confusing test failure messages.
(A lot of the test templates in this file have this issue.)  

The [signature for assert_equal](http://docs.ruby-lang.org/en/2.1.0/MiniTest/Assertions.html#method-i-assert_equal) is: `assert_equal(exp, act, msg = nil)`